README.md

```diff
@@ -32,7 +35,94 @@ Since the project is under active development, we recommend watching the reposit
 regularly for updates. Detailed documentation and usage instructions will be provided as the project
 progresses.
 
-## Testing the Mender MCU Client (POSIX)
+
+## Zephyr OS integration
+
+We have created a reference application to integrate with Mender MCU. It can be found in the
+repository [mender-mcu-integration](https://github.com/mendersoftware/mender-mcu-integration). The
+policy is that the reference application's `main` branch will follow the latest stable branch of
+`mender-mcu` (this repository).
+
+In the `mender-mcu-repository` you will find:
+* A reference `west` workspace for Zephyr OS builds, including `CMake` and `KConfig` configurations.
+* Sample application code that for how to use `mender-mcu` APIs.
+* A list of boards that we have so far tested in Northern.tech.
+
+We recommend to start there, to get a feeling on how we have designed our solution, and then come
+back here for more in-depth information.
+
+
+### The Mender client API
+
+The main API is exposed in [`mender-client.h`](include/mender-client.h). The most notable parts are:
+
+* Identity callback. The user needs to provide a callback that will return the identity. In most
+cases this would be reading a unique number of the device like a MAC address. Read more about
+[Mender identity in Mender Docs](https://docs.mender.io/overview/identity).
+
+* Network connect / Network release callbacks. In systems where the network is not always available,
+the client will invoke the network connect callback before doing operations that require
+connectivity (for example, checking for an update). Once the network is done, network release
+callback will be invoked. They can be set to `NULL` if the network is assumed to be always ready.
+
+* Restart callback. To be called on restart, for example after a fatal error.
+
+* Inventory setting. There is an explicit call to set the inventory. This will update the internal
+inventory of the client and be reported in the next inventory interval.
+
+See the source code for more.
+
+
+### Update Modules API
+
+Mender MCU models the Update Modules API that we have been using in the regular Mender client for
+years. Refer to the [Update Modules chapter in Mender
+Docs](https://docs.mender.io/artifact-creation/create-a-custom-update-module) for an introduction.
+In Mender MCU, the modules are not "executables" but rather a piece of code that is integrated with
+the client to handle each type of update.
+
+#### Update Modules State Machine
+
+As stated above, an Update Module for Mender MCU is just C code. In particular, it is defined by a C
```

Comment:
is a customizable C function ("just" sounds like it's being marginalized)

Concretely, once created, all Update Modules are identified by a C struct ...